src/sqlfluff/core/parser/parsers.py

```diff
@@ -100,6 +100,47 @@ def match(
         return MatchResult.from_unmatched(segments)
 
 
+class MultiStringParser(StringParser):
+    """An object which matches and returns raw segments on a collection of strings."""
+
+    def __init__(
+        self,
+        templates: Collection[str],
+        raw_class: Type[RawSegment],
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        optional: bool = False,
+        **segment_kwargs,
+    ):
+        self.templates = {template.upper() for template in templates}
+        super().__init__(
+            template="",
+            raw_class=raw_class,
+            name=name,
+            type=type,
+            optional=optional,
+            **segment_kwargs,
+        )
+        # Delete attribute which is replaced by `self.templates` for this `Parser``
+        del self.template
+
+    def simple(self, parse_context: "ParseContext") -> Optional[List[str]]:
+        """Return simple options for this matcher.
+
+        Because string matchers are not case sensitive we can
+        just return the templates here.
+        """
+        return list(self.templates)
```

Comment:
Is this function called frequently? I notice that we're creating and returning a new list each time, which could be slow.

Reply:
The compromise here was whether to cache the `set` which is used for matching or to cache the `list` which is expected to be returned by `simple`. We could cache both as class attributes?

Reply:
I've implemented a `self._simple` which is set in `__init__` which should avoid creating the `list` multiple times (this goes for `StringParser` too).